ssl/statem/statem_clnt.c

```diff
@@ -1130,12 +1133,27 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             }
         }
 
+        /* Fix session version if it is not reused and we are not in EAP-FAST */
+        if (s->session->tlsext_tick == NULL)
```

Comment:
Are you sure you don't want this to run in EAP-FAST too? I think the old line 979 would run everywhere. Then you don't need this slightly confusing check.